platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformUpstreamDependenciesResolver.java

```diff
@@ -131,8 +135,49 @@ public DefaultTransformUpstreamDependenciesResolver(
         this.requestAttributes = requestAttributes;
         this.artifactDependencySortOrder = artifactDependencySortOrder;
 
-        this.resolverResults = resolverResults;
-        this.strictResolverResults = strictResolverResults;
+        this.initialVisitedArtifacts = partialVisitedArtifacts;
+        this.initialVisitedGraph = partialVisitedGraph;
+        this.completeGraphResults = calculatedValueContainerFactory.create(Describables.of("complete graph results for", resolutionHost.getDisplayName()), context -> {
+            // TODO: We should acquire the project lock here, since this will resolve a configuration, which requires a project lock.
```

Comment:
I'm not sure if there is a reason we are not. We didn't in the past and I didn't really want to change the behavior in these changes.

I wanted to leave a note that we are not doing it though to at least document it. I think this whole class needs rework since its really confusing imo

Reply:
Further, this is really something that should be handled by the `CalculatedValue` infrastructure. `ValueCalculator` has the option to specify a `getOwningProject` and `usesMutableProjectState`, so preferably the requirements of `fullGraphResults.finalizeIfNotAlready` (which locks to acquire) would be encoded when we create `fullGraphResults` and not when we consume `fullGraphResults`